hcl2template/types.packer_config.go

```diff
@@ -122,12 +124,22 @@ func (cfg *PackerConfig) EvalContext(ctx BlockContext, variables map[string]cty.
 
 	iterID, ok := cfg.HCPVars["iterationID"]
 	if ok {
+		log.Printf("[WARN] Deprecation: Contextual Variable `iterationID` has been deprecated packer context. " +
+			"Please use `versionFingerprint` variable instead.")
 		ectx.Variables[packerAccessor] = cty.ObjectVal(map[string]cty.Value{
 			"version":     cty.StringVal(cfg.CorePackerVersionString),
 			"iterationID": iterID,
 		})
 	}
 
+	versionFingerprint, ok := cfg.HCPVars["versionFingerprint"]
```

Comment:
That will make `iterationID` unusable in this state unfortunately, since we create the object in this branch, and we do the same above, so this assignation will erase the previous value.

I would suggest changing a couple of things here, the previous approach is probably not valid anymore.
Let's remove the creation of the `packerAccessor` object from the declaration of the context, we can build it along the way, and only create it when it is in its final state.

```go
packerVars := map[string]cty.Value{
	"version": cty.StringVal(cfg.CorePackerVersionString),
	"iterationID": cty.UnknownVal(cty.String),
	"versionFingerprint": cty.UnknownVal(cty.String),
}

iterID, ok := cfg.HCPVars["iterationID"]
if ok {
	packerVars["iterationID"] = cty.StringVal(iterID)
}
versionFP, ok := cfg.HCPVars["versionFingerprint"]
if ok {
	packerVars["versionFingerprint"] = cty.String(versionFP)
}

ectx.Variables[packerAccessor] = cty.ObjectVal(packerVars)
```